packages/fx-core/src/component/driver/env/appsettingsGenerate.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as fs from "fs-extra";
+import { Service } from "typedi";
+
+import { hooks } from "@feathersjs/hooks/lib";
+import { FxError, Result, SystemError, UserError } from "@microsoft/teamsfx-api";
+
+import { getLocalizedString } from "../../../common/localizeUtils";
+import { wrapRun } from "../../utils/common";
+import { logMessageKeys } from "../aad/utility/constants";
+import { DriverContext } from "../interface/commonArgs";
+import { StepDriver } from "../interface/stepDriver";
+import { addStartAndEndTelemetry } from "../middleware/addStartAndEndTelemetry";
+import { InvalidParameterUserError } from "./error/invalidParameterUserError";
+import { UnhandledSystemError } from "./error/unhandledError";
+import { GenerateAppsettingsArgs } from "./interface/generateAppsettingsArgs";
+
+const actionName = "appsettings/generate";
+const helpLink = "https://aka.ms/teamsfx-actions/appsettings-generate";
+
+@Service(actionName) // DO NOT MODIFY the service name
+export class GenerateAppsettingsDriver implements StepDriver {
+  @hooks([addStartAndEndTelemetry(actionName, actionName)])
+  public async run(
+    args: GenerateAppsettingsArgs,
+    context: DriverContext
+  ): Promise<Result<Map<string, string>, FxError>> {
+    return wrapRun(() => this.handler(args, context));
+  }
+
+  private async handler(
+    args: GenerateAppsettingsArgs,
+    context: DriverContext
+  ): Promise<Map<string, string>> {
+    try {
+      this.validateArgs(args);
+
+      const appSettingsJson = JSON.parse(fs.readFileSync(args.target, "utf-8"));
```

Comment:
Any reusable logic from `fx-core/src/component/code/appSettingsUtils`?

Reply:
For now, there are few reusable parts. ```appSettingsUtils``` use scenario and regex to replace ```$$``` parts. This code will replace ```appsettings.Development.json``` all parts which are provided in ```local.yml``` and remain the parts not provided.